core/store/src/trie/mod.rs

```diff
@@ -1675,15 +1660,6 @@ impl Trie {
                     for (node_hash, serialized_node) in trie_accesses.nodes {
                         recorder.borrow_mut().record(&node_hash, serialized_node);
                     }
-                    for (value_hash, value) in trie_accesses.values {
```

Comment:
are these only deletions or also include values updated (ie. do you mean by `deleted values` the values being replaced with other values or deleted)? is the reason this is safe because the validator will run the code that will provide this value anyways?

Reply:
No, these are all deleted values. Let me explain that:

If you are validator, you trust to the root first, because it was signed by validators on the previous step. And then you trust its children, because they hash into the root and we have uniqueness, then we trust their children, etc.
And when we reach the ValueRef, we don't actually need the whole value. We need `ValueRef::length` to recompute memory usage, which we trust already, and after that we just set value to None and squash necessary nodes on the way.

Intermediate nodes are still required to prove that value exists, though.

Side note - inserted values are not part of the proof, because they are result of previous state proof + chunk execution.

Reply:
got it, thanks for explaining. lgtm